mockgen/package_mode.go

```diff
@@ -0,0 +1,358 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"go/types"
+	"strings"
+
+	"go.uber.org/mock/mockgen/model"
+	"golang.org/x/tools/go/packages"
+)
+
+var (
+	buildFlags = flag.String("build_flags", "", "(package mode) Additional flags for go build.")
+)
+
+type packageModeParser struct {
+	pkgName string
+}
+
+func (p *packageModeParser) parsePackage(packageName string, ifaces []string) (*model.Package, error) {
+	p.pkgName = packageName
+
+	pkg, err := p.loadPackage(packageName)
+	if err != nil {
+		return nil, fmt.Errorf("load package: %w", err)
+	}
+
+	interfaces, err := p.extractInterfacesFromPackage(pkg, ifaces)
+	if err != nil {
+		return nil, fmt.Errorf("extract interfaces from package: %w", err)
+	}
+
+	return &model.Package{
+		Name:       pkg.Types.Name(),
+		PkgPath:    packageName,
+		Interfaces: interfaces,
+	}, nil
+}
+
+func (p *packageModeParser) loadPackage(packageName string) (*packages.Package, error) {
+	var buildFlagsSet []string
+	if *buildFlags != "" {
+		buildFlagsSet = strings.Split(*buildFlags, " ")
+	}
+
+	cfg := &packages.Config{
+		Mode:       packages.NeedDeps | packages.NeedImports | packages.NeedTypes | packages.NeedTypesInfo | packages.NeedEmbedFiles,
```

Comment:
@tulzke / @JacobOaks any particular reason that `NeedDeps` was used here? I was looking into mock generation performance and this seems to cause a significant slow down (as expected) as all dependencies are parsed.

With the option removed no tests are breaking and for us its working without, but maybe you had a case in mind that needs `NeedDeps` https://github.com/uber-go/mock/pull/230